src/psi_io.f90

```diff
@@ -699,9 +702,10 @@ subroutine rdhdf_2d (fname,scale,nx,ny,f,x,y,ierr)
             scale=s%scale
             ! x=>s%scales(1)%f
             ! y=>s%scales(2)%f
-
-            x=s%scales(1,1)
-            y=s%scales(2,1)
+            ! [XX: Allocate First and then copy the elements of s%scales to x and y dimensions]
+            allocate(x(nx), y(ny))
+            x = s%scales(1,1:nx)
```

Comment:
Ah, I see. this is nice, I was trying it myself as well, and this is where it's being assigned. I missed this,

This is nice.